test/inductor/test_cooperative_reductions.py

```diff
@@ -0,0 +1,115 @@
+# Owner(s): ["module: inductor"]
+import torch
+import torch._inductor
+from torch._inductor import config
+from torch._inductor.utils import run_and_get_code
+from torch.testing._internal.common_utils import (
+    instantiate_parametrized_tests,
+    parametrize,
+    TestCase,
+)
+from torch.testing._internal.inductor_utils import HAS_CUDA
+
+
+@config.patch(
+    {
+        "triton.cooperative_reductions": True,
+        "triton.force_cooperative_reductions": True,
+    }
+)
+@instantiate_parametrized_tests
+class CooperativeReductionTests(TestCase):
+    def setUp(self):
+        super().setUp()
+        torch._inductor.metrics.generated_kernel_count = 0
+        torch._dynamo.reset()
+
+    def run_and_check(self, fn, args, *, expect_kernel_count=1):
```

Comment:
It looks like you're handling this correctly already but - nit: add tests for fp16, fp64 ?